addons/website_slides/tests/test_slide_channel.py

```diff
@@ -404,6 +404,15 @@ def test_channel_enroll_policy(self):
         copied_channel = channel.copy()
         self.assertEqual(copied_channel.enroll, 'invite', "Copied channel should have the same enroll field value")
 
+    @users('user_officer')
+    def test_duplicate_channel_preserves_sequence(self):
+        self.slide_3.write({'sequence': 1})
+        self.slide.write({'sequence': 4})
+        self.channel.invalidate_recordset()
+        copied_channel = self.channel.copy()
+
+        self.assertEqual(copied_channel.slide_ids.sorted('sequence')[0].name, self.slide_3.name)
```

Comment:
Hello @Ephoniz , 

Thanks for the explanation yesterday. 

I think the explanation in your commit/PR is not aligned with your solution anymore as the problem comes from the copy_data that set the sequence of all slide to 0 which is breaking the slide order (fallback to id order as all sequence are equals):
"After the changes applied to improve the batch call of copy in ... the values that we get after the copy are ordered by id instead of preserving the original order, so for this case we can make sure to reorder the values and mantain the proper sequence when duplicating inside copy_data()."

Maybe you can change it with something like this:
How to reproduce:
- Install website_slides
- Start the shell
- type "channel = env['slide.channel'].search([])[0]" (Basics of Gardening)
- then "channel.slide_ids.mapped('sequence')"
- You get [1, 2, 3, 4, 5]
- But when typing "channel.copy().slide_ids.mapped('sequence')"
- You get [0, 0, 0, 0, 0]

So the slide order of the course is lost when copying the channel.

Assigning the sequence to 0 is useful when copying one slide so that it appears
at the top after the copy. So to keep that feature and preserve the sequence 
order of the slides when copying a channel, we only assign to 0 the sequence if
we are not copying a channel by checking that __copy_data_seen doesn't contains
slide.channel. 
The context variable "__copy_data_seen" is used to: "
avoid recursion through already copied records in case of circular relationship
". So when copying a channel it contains the slide channel being copied (model
and id as a map).


In the test, you can maybe use the same technic to compare the sequence:
```python
self.assertEqual(
    self.channel.slide_ids.mapped('sequence'), 
    self.channel.copy().slide_ids.mapped('sequence'),
    "sequence are preserved"
)
```

Also I would add a test that checks that when copying slides alone, the sequence is reset to 0.